### DISCUSSION TITLE
Scheduler.GetCurrentlyExecutingJobs() always empty

### DISCUSSION THREAD

**Discussion by ITaluone** [quartznet/quartznet#1723] [Q&A] [answered]
Hi

I am trying to build an ASP.NET Core 6 web api project which controls jobs. 
I add the quartz service like the following:

## My setup
```csharp
builder.Services.AddQuartz(conf =>
{
    conf.InterruptJobsOnShutdown = true;
    conf.UseInMemoryStore();
    conf.UseMicrosoftDependencyInjectionJobFactory();

    conf.SchedulerId = "MyId";
    conf.SchedulerName = "MyName";
    conf.UseDefaultThreadPool(15);
    conf.UseSimpleTypeLoader();
    conf.UseTimeZoneConverter();
});

builder.Services.AddQuartzServer(conf =>
{
    conf.WaitForJobsToComplete = false;
});
```

And in my controller I schedule a job like this:

```csharp
public class ExecutionController : Controller {
    private readonly ISchedulerFactory _factory;

    public ExecutionController(ISchedulerFactory factory)
    {
        _factory = factory;
    }

    [HttpGet]
    public async Task<bool> Execute()
    {
        var scheduler = await _factory.GetScheduler();

        if (!scheduler.IsStarted)
            await scheduler.Start();

        var jobKey = CreateJobKey();

        if (await scheduler.CheckExists(jobKey))
            return false;

        var job = JobBuilder.Create<MyJob>()
            .WithDescription($"Executing")
            .WithIdentity(jobKey)
            .Build();

        var trigger = TriggerBuilder.Create()
            .WithIdentity(setting.Id)
            .StartNow()
            .WithCronSchedule("*/10 * * * * ?")
            .EndAt(DateTime.Now.AddSeconds(20))
            .Build();

        await scheduler.ScheduleJob(job, trigger);

        return true;
    }
}
```

## My problem
I want periodically check if the job is running and get some status via an endpoint:

```csharp
    [HttpGet("Status")]
    public async Task<IActionResult> GetStatus()
    {
        var jobKey = GetJobKey();

        var scheduler = await _factory.GetScheduler();

        var executingJobs = await scheduler.GetCurrentlyExecutingJobs(); // <--- this statement always returns an empty list
        var executingJob = executingJobs.FirstOrDefault(job => job.JobDetail.Key.Name == jobKey.Name);

        return Ok(new StatusViewModel { JobName = job?.Key.Name });
    }
```

What I am doing wrong here?

Thanks for your advice.

**Comment by lahma** [comment, marked as answer]
How long does it take for you job to execute? `GetCurrentlyExecutingJobs()` returns jobs that are currently executing, not all jobs that have been registered to scheduler.

**Reply by ITaluone** [reply]
This was only a example. If my trigger ends in one hour the behaviour stays the same.

**Reply by ITaluone** [reply]
Ahh.. my job executes immediately.. My job just prints the actual `DateTime.Now`. This works, but `GetCurrentlyExecutingJobs()`  returns an empty list.

**Reply by lahma** [reply]
Trigger != job. The method returns *running* jobs, like ones actively executing the job execution logic.

**Reply by ITaluone** [reply]
I have made a small screencast to show the problem in action:



[2022-08-24_15-39-39.zip](https://github.com/quartznet/quartznet/files/9416617/2022-08-24_15-39-39.zip)

**Reply by lahma** [reply]
Sorry, can't open this.

**Reply by IT-VBFK** [reply]
[screencast.m4v.zip](https://github.com/quartznet/quartznet/files/9418975/screencast.m4v.zip)

Hope this fits better :)

**Reply by IT-VBFK** [reply]
>  like ones actively executing the job execution logic.

Does this mean that a job whose execution logic only is writing the current date time into console is too fast to get it with this method?

If so.. is there a alternative approach to get all jobs which are started and "running" (meaning trigger is started and  between start and end)?

**Reply by lahma** [reply]
trigger != job.  if you start a trigger that takes couple millisends to run (Console.WriteLine) it's hard to time your `GetCurrentlyExecutingJobs()` in a way that captures any running job. Think of postal service: "deliver every day between 6-8am" there are no deliveries probably active at 11am even though the agreement is from January to December. Triggers fire all year, but jobs are active during delivery.

**Reply by ITaluone** [reply]
Ok. Now I got it :)

Thanks for your help
